### ISSUE TITLE
Extend Docs Samples Pipeline - Allow Transformation Script

### ISSUE THREAD

**Issue by scbedd** [Azure/azure-sdk#1193]
We are looking for a way to deal with [this issue](https://github.com/Azure/azure-sdk-for-net/issues/10937) generically.

tl;dr for above issue. The csproj zipped up in the sample will build from within the repository, but will not build from outside of it. There are project references present. This means that it won't built from within the isolated samples zip.

Currently the docs samples pipeline has zero interaction with the code. It clones it down, looks for READM.md, scans for identifiable front-matter, and zips up all files specified by said front-matter.

@jsquire asks that we add custom script invocation just prior to the zipping up of the code. For instance, in the yaml front-matter, we would do something like the following:

```
---
page_type: sample
languages:
- csharp
products:
- azure
- azure-event-hubs
name: Azure.Messaging.EventHubs samples for .NET
description: Samples for the Azure.Messaging.EventHubs client library
extensionScript: ../transform.ps1
---
```

Leveraging a feature like this would allow us to transform the only-locally-buildable csproj to one that can be built in isolation. This would also allow us to maintain a single version of the csproj at rest in the repository.
